build-tools/packages/build-cli/docs/typetestDetails.md

```diff
@@ -0,0 +1,151 @@
+The `generate typetests` command generates type compatibility tests based on the individual package settings in
+package.json.
+
+## What are type compatibility tests?
+
+Type compatibility tests are automatically generated test cases that check for type incompatibilities across package
+versions. They work by comparing types in the current version to the types in a previous version. The generated tests
+use the previous type in place of the new type, and vice-versa.
+
+See the [API Type Validation][] page in the Fluid Framework wiki for more information about type compatibility tests,
+including how to annotate expected type incompatibilities to fix failing tests.
+
+## Configuring type tests
+
+Type test generation is primarily configured in the package.json file of the package being tested, in the
+`typeValidation` section:
+
+```json
+"typeValidation": {
+  "disabled": false,
+  "version": "1.1.0",
+  "broken": {
+    "InterfaceDeclaration_AzureClientProps": {
+      "forwardCompat": false
+    }
+  }
+}
+```
+
+The `broken` section is used to indicate known breaking changes. Type tests can be completely disabled for a package
+using the `disabled` property. See [API Type Validation][] for more information.
+
+Generating type tests has two parts: _preparing package.json_ and _generating test modules_. By default, both steps are
+run for each package. You can run only one part at a time using the `--prepare` and `--generate` flags.
+
+[api type validation]: https://github.com/microsoft/FluidFramework/wiki/API-Type-Validation
+
+## Preparing package.json
+
+_Preparing package.json_ determines the baseline previous version to use, then sets that version in package.json. If the
+previous version changes after running preparation, then `npm install` must be run before the generate step will run
+correctly.
+
+Optionally, any type tests that are marked "broken" in package.json can be reset using the `--reset` flag during
+preparation. This is useful when resetting the type tests to a clean state, such as after a major release.
+
+## Generating tests
+
+Generating test modules takes the type test information from package.json, most notably any known broken type tests, and
+generates test files that should be committed. By default, the generated files will contain `.generated` in their name,
+but this can be suppressed with the `--no-generateInName` flag.
+
+## Branch configuration
+
+Type tests can be configured to use different baseline versions on a given branch depending on the type of release that
+the branch is designated for. For example, for the client release group, the _next_ branch is the _major version series
+branch_ and _main_ is the _minor version series branch_. This can be declared in the root package.json, in the
+`fluidBuild.repoPackages` section. For example, the following configuration designates the _main_ and _lts_ branches as
+minor version series branches, while the _next_ branch is designated for major releases.
+
+```json
```

Comment:
Would it be possible to specify this in each release group's package json, rather than specifying all of the release groups in this package.json file? This pattern feels strange since the root package.json is really the client release group's, not the repo's.

Reply:
I put it in the root because that's where all the release groups are defined. Our structure right now means that the root package.json serves two purposes: it's both the client release group package.json and repo-wide config.